app/src/org/commcare/adapters/EntityStringFilterer.java

```diff
@@ -121,7 +121,8 @@ private void buildMatchList() {
                     matchScores.add(Pair.create(index, score));
                 }
             }
-            if (isAsyncMode) {
+            // If fuzzy search is enabled need to re-sort based on edit distance
+            if (isFuzzySearchEnabled) {
```

Comment:
@wpride Why would we only want to do this when fuzzy search is enabled? It looks like the `matchScores` object gets populated whether or not we're using fuzzy search, so wouldn't we just always want to sort by those scores?

Reply:
Because if fuzzy search isn't enabled then all we have is the set of entries matching the search fields exactly, so I think in that case keeping the default sort is correct